lib/symbol-provider.js

```diff
@@ -0,0 +1,371 @@
+      return editorSubscriptions.dispose()
+    }))
+
+    bufferEditors = this.watchedBuffers.get(buffer)
+    if (bufferEditors) {
+      return bufferEditors.push(editor)
+    } else {
+      let bufferSubscriptions = new CompositeDisposable()
+      if (bufferSupportsStopChanging()) {
+        bufferSubscriptions.add(buffer.onDidStopChanging(({changes}) => {
+          let editors = this.watchedBuffers.get(buffer)
+          if (editors && editors.length && editors[0]) {
+            return changes.map(({start, oldExtent, newExtent}) =>
+              this.symbolStore.recomputeSymbolsForEditorInBufferRange(editors[0], start, oldExtent, newExtent))
+          }
+        }))
+      } else {
+        bufferSubscriptions.add(buffer.onDidChange(({oldRange, newRange}) => {
+          let editors = this.watchedBuffers.get(buffer)
+          if (editors && editors.length && editors[0]) {
+            let { start } = oldRange
+            let oldExtent = oldRange.getExtent()
+            let newExtent = newRange.getExtent()
+            return this.symbolStore.recomputeSymbolsForEditorInBufferRange(editors[0], start, oldExtent, newExtent)
+          }
+        }))
+      }
+
+      bufferSubscriptions.add(buffer.onDidDestroy(() => {
+        this.symbolStore.clear(buffer)
+        bufferSubscriptions.dispose()
+        return this.watchedBuffers.delete(buffer)
+      }))
+
+      this.watchedBuffers.set(buffer, [editor])
+      return this.buildWordListOnNextTick(editor)
+    }
+  }
+
+  isWatchingEditor (editor) {
+    return this.getWatchedEditorIndex(editor) > -1
+  }
+
+  isWatchingBuffer (buffer) {
+    return (this.watchedBuffers.get(buffer) != null)
+  }
+
+  getWatchedEditorIndex (editor) {
+    let editors = this.watchedBuffers.get(editor.getBuffer())
+    if (editors) {
+      return editors.indexOf(editor)
+    } else {
+      return -1
+    }
+  }
+
+  updateCurrentEditor (currentPaneItem) {
+    if (currentPaneItem == null) { return }
+    if (currentPaneItem === this.editor) { return }
+    this.editor = null
+    if (this.paneItemIsValid(currentPaneItem)) {
+      this.editor = currentPaneItem
+      return this.editor
+    }
+  }
+
+  buildConfigIfScopeChanged ({editor, scopeDescriptor}) {
+    if (!this.scopeDescriptorsEqual(this.configScopeDescriptor, scopeDescriptor)) {
+      this.buildConfig(scopeDescriptor)
+      this.configScopeDescriptor = scopeDescriptor
+      return this.configScopeDescriptor
+    }
+  }
+
+  buildConfig (scopeDescriptor) {
+    this.config = {}
+    let legacyCompletions = this.settingsForScopeDescriptor(scopeDescriptor, 'editor.completions')
+    let allConfigEntries = this.settingsForScopeDescriptor(scopeDescriptor, 'autocomplete.symbols')
+
+    // Config entries are reverse sorted in order of specificity. We want most
+    // specific to win; this simplifies the loop.
+    allConfigEntries.reverse()
+
+    for (let i = 0; i < legacyCompletions.length; i++) {
+      let {value} = legacyCompletions[i]
```

Comment:
Change `let` to `const`. Also add some whitespace inside the brackets to be consistent with other, similar code?